model/map/src/main/java/org/keycloak/models/map/realm/AbstractRealmEntity.java

```diff
@@ -0,0 +1,1030 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.realm;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.keycloak.common.util.Time;
+import org.keycloak.models.ModelDuplicateException;
+import org.keycloak.models.OTPPolicy;
+import org.keycloak.models.map.common.AbstractEntity;
+import org.keycloak.models.map.realm.entity.MapAuthenticationExecutionEntity;
+import org.keycloak.models.map.realm.entity.MapAuthenticationFlowEntity;
+import org.keycloak.models.map.realm.entity.MapAuthenticatorConfigEntity;
+import org.keycloak.models.map.realm.entity.MapClientInitialAccessEntity;
+import org.keycloak.models.map.realm.entity.MapComponentEntity;
+import org.keycloak.models.map.realm.entity.MapIdentityProviderEntity;
+import org.keycloak.models.map.realm.entity.MapIdentityProviderMapperEntity;
+import org.keycloak.models.map.realm.entity.MapOTPPolicyEntity;
+import org.keycloak.models.map.realm.entity.MapRequiredActionProviderEntity;
+import org.keycloak.models.map.realm.entity.MapRequiredCredentialEntity;
+import org.keycloak.models.map.realm.entity.MapWebAuthnPolicyEntity;
+
+public abstract class AbstractRealmEntity<K> implements AbstractEntity<K> {
+
+    private final K id;
+    private String name;
+
+    private Boolean enabled = false;
```

Comment:
I can see the need for adding the default values here for the time being, and it makes perfect sense as of now. This is just a note that these default values will eventually go away to allow for inheriting the defaults from a parent realm when the value would be `null`.